### PR TITLE
lxd/db/generate/db/stmt: Add joinon tag

### PR DESCRIPTION
**Pull request by masnax**
Allows specifying the column name on which to join on. 

Works like this:

```go
type Service struct {
        ID                  int
	MemberID            int
	Member              string `db:"primary=yes&join=internal_cluster_members.name-ON-member_id"`
	Service             string `db:"primary=yes"`
}
```

### CODE REVIEW

**Comment by tomponline** [comment]
@masnax is there a readme file that you can document this (and the existing) `db:` tags?

**Comment by tomponline** [comment]
@masnax in the example you gave here, why is MemberID also not joined?

**Review by tomponline — APPROVED** [review]
Would be great to get these tags documented.

**Comment by tomponline** [comment]
@masnax would it be safer/more flexible to introduce a `joinon` tag, e.g.

```go
type Service struct {
        ID                  int
	MemberID            int
	Member              string `db:"primary=yes&join=internal_cluster_members.name&joinon=member_id"`
	Service             string `db:"primary=yes"`
}
```
That way no risk of conflicting with the field name.

File: lxd/db/generate/db/stmt.go [hunk 1]
```diff
@@ -213,12 +218,17 @@ func (s *Stmt) objects(buf *file.Buffer) error {
 		}
 
 		right := strings.Split(join, ".")[0]
+		joinOn := field.Config.Get("leftjoinon")
```
Comment (tomponline) [inline lxd/db/generate/db/stmt.go:221]: We dont need leftjoinon, because its implicit when using `leftjoin` tag.

**Review by tomponline — APPROVED** [review]
LGTM.

**Comment by masnax** [comment]
Ah actually wait, saw a bug

**Comment by masnax** [comment]
Fixed now. Just a string typo.

**Pull request merged**